app/controllers/users_controller.rb

```diff
@@ -0,0 +1,61 @@
+class UsersController < ApplicationController
+
+  get "/signup" do
+    if !logged_in?
+      erb :signup
+    else
+      redirect to "/tweets"
+    end
+  end
+
+  get "/login" do
+    if !logged_in?
+      erb :login
+    else
+      redirect to "/tweets"
+    end
+  end
+
+  post "/signup" do
+    if params[:username] == "" || params[:email] == "" || params[:password] == ""
+      flash[:message] = "All fields are required."
+      redirect to "/signup"
+    else
+      user = User.new(username: params[:username], email: params[:email], password: params[:password])
+      if user.save
+        session[:id] = user.id
```

Comment:
your current_user method is using session[:user_id] these should all be the same